### PR TITLE
Update dependency go modules for k8s v1.28.0

### PR DESCRIPTION
**Pull request by Sneha-at**
Ran kubernetes-csi/csi-release-tools go-get-kubernetes.sh -p 1.28.0

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -6,7 +6,7 @@ require (
 	github.com/kubernetes-csi/csi-lib-utils v0.15.0-rc0.0.20230811202321-0f6938005f63
 	golang.org/x/sys v0.11.0
 	google.golang.org/grpc v1.57.0
-	k8s.io/client-go v0.28.0-rc.0
+	k8s.io/client-go v0.28.0
 	k8s.io/klog/v2 v2.100.1
 	k8s.io/kubelet v0.28.0-rc.0
```
Comment (jsafrane) [inline go.mod:11]: This should be v0.28.0

Comment (Sneha-at) [reply go.mod]: Updated the version

File: go.mod [hunk 2]
```diff
@@ -6,9 +6,9 @@ require (
 	github.com/kubernetes-csi/csi-lib-utils v0.15.0-rc0.0.20230811202321-0f6938005f63
```
Comment (sunnylovestiramisu) [inline go.mod:6]: This is still rc,  we need to update when csi-lib-utils v0.15.0 is ready

Comment (Sneha-at) [reply go.mod]: Updated `csi-lib-utils v0.15.0`

**Pull request merged**